src/nccl_ofi_net.c

```diff
@@ -1240,7 +1240,11 @@ static ncclResult_t ofi_getProperties(int dev, ncclNetProperties_t *props)
 	/* name is NULL if device is a part of multirail config */
 	if (nic_info->device_attr->name) {
 		dev_props.name = strdup(nic_info->device_attr->name);
+	} else {
+		dev_props.name = malloc(NAME_MAX);
+		snprintf(dev_props.name, NAME_MAX, "%s, device %d", nic_prov->domain_attr->name, dev);
```

Comment:
TBH, this isn't required as we do set the defaults in `set_nic_props_default`. So it should be enough to skip setting a new name if the provider didn't provide a specific device name. Could you please revert back to your first revision and add a comment about overriding only if the value is available?

Sorry for the inconvenience!

Reply:
Force-pushed squashed version with request addressed